services/continuousquery/service.go

```diff
@@ -56,48 +56,55 @@ type Service struct {
 		Databases() map[string]*meta.DatabaseInfo
 	}
 
-	mu             sync.RWMutex
-	lastRuns       map[string]time.Time
-	QueryExecutor  *query.Executor
-	ReportInterval time.Duration
+	mu                 sync.RWMutex
+	lastRuns           map[string]time.Time
+	QueryExecutor      *query.Executor
+	ReportInterval     time.Duration
+	MaxProcessCQNumber int
+	ContinuousQuery    []*ContinuousQuery
 }
 
 // NewService creates a new Service instance named continuousQuery
-func NewService(interval time.Duration) *Service {
+func NewService(interval time.Duration, number int) *Service {
 	s := &Service{
-		lastRuns:       map[string]time.Time{},
-		ReportInterval: DefaultReportTime,
-		QueryExecutor:  query.NewExecutor(),
+		lastRuns:           map[string]time.Time{},
+		QueryExecutor:      query.NewExecutor(),
+		ReportInterval:     DefaultReportTime,
+		MaxProcessCQNumber: number,
 	}
 	s.Init("continuousQuery", interval, s.handle)
 	return s
 }
 
 func (s *Service) handle() {
-	if !s.hasContinuousQuery() {
+	s.ContinuousQuery = []*ContinuousQuery{}
```

Comment:
try to reuse memory:
```suggestion
	s.ContinuousQuery = s.ContinuousQuery[:0]
```